src/main/java/io/cdap/plugin/servicenow/apiclient/ServiceNowTableAPIClientImpl.java

```diff
@@ -375,45 +238,146 @@ public List<Map<String, Object>> fetchTableRecordsRetryableMode(String tableName
+      requestBuilder.setEntity(entity);
+      apiResponse = executePost(requestBuilder.build());
+
+      systemID = String.valueOf(getSystemId(apiResponse));
+
       if (!apiResponse.isSuccess()) {
         LOG.error("Error - {}", getErrorMessage(apiResponse.getResponseBody()));
-        collector.addFailure("Unable to fetch schema for table " + tableName, null).
-          withConfigProperty(ServiceNowConstants.PROPERTY_TABLE_NAME);
-        return null;
-      }
-      List<SchemaResponse> result = parseSchemaResponse(apiResponse.getResponseBody());
-      List<ServiceNowColumn> columns = new ArrayList<>();
-
-      if (result != null && !result.isEmpty()) {
-        for (SchemaResponse field : result) {
-          columns.add(new ServiceNowColumn(field.getName(), field.getInternalType()));
-        }
+      } else {
+        LOG.info(apiResponse.getResponseBody());
       }
-      return SchemaBuilder.constructSchema(tableName, columns);
-    } catch (OAuthSystemException | OAuthProblemException e) {
-      LOG.error("Error in connection - {}", e);
-      collector.addFailure("Connection failed. Unable to fetch schema for table " + tableName, null);
+    } catch (OAuthSystemException | OAuthProblemException | UnsupportedEncodingException e) {
+      LOG.error("Error in creating a new record", e);
+      throw new RuntimeException("Error in creating a new record");
     }
-    return null;
+    return systemID;
   }
 
-  @VisibleForTesting
-  public List<SchemaResponse> parseSchemaResponse(String responseBody) {
-    JsonObject jo = gson.fromJson(responseBody, JsonObject.class);
-    JsonArray ja = jo.getAsJsonArray(ServiceNowConstants.RESULT);
-    Type type = new TypeToken<List<SchemaResponse>>() {
-    }.getType();
-    return gson.fromJson(ja, type);
+  private String getSystemId(RestAPIResponse apiResponse) {
+    JsonObject jsonObject = gson.fromJson(apiResponse.getResponseBody(), JsonObject.class);
+    JsonObject result = (JsonObject) jsonObject.get(ServiceNowConstants.RESULT);
+
+    return result.get(ServiceNowConstants.SYSTEM_ID).getAsString();
   }
 
+  /**
+   * This function is being used in end-to-end (e2e) tests to fetch a record
+   * Return a record from ServiceNow application.
+   *
+   * @param tableName The ServiceNow table name
+   * @param query     The query
+   */
+  public Map<String, String> getRecordFromServiceNowTable(String tableName, String query)
+    throws OAuthProblemException, OAuthSystemException {
+
+    ServiceNowTableAPIRequestBuilder requestBuilder = new ServiceNowTableAPIRequestBuilder(
+      this.conf.getRestApiEndpoint(), tableName, false)
+      .setQuery(query);
+
+    RestAPIResponse apiResponse = null;
+    String accessToken = getAccessToken();
+    requestBuilder.setAuthHeader(accessToken);
+    apiResponse = executeGet(requestBuilder.build());
+    JsonObject jsonObject = gson.fromJson(apiResponse.getResponseBody(), JsonObject.class);
```

Comment:
similar comment as elsewhere, use concrete java classes and not JsonObject

Reply:
Added APIResponse class and deserialized the response directly to the Java object. a963c57e661447a3f1a6e970cdabc6f76964b313